### PR TITLE
shellcheck fixes

### PR DESCRIPTION
**Pull request by stevestock**
Many changes per [ShellCheck](https://github.com/koalaman/shellcheck).  Please review, a lot of changes, and I don't see a test suite available.

Notes:
dename has been unused since d173597a065fea81f1f0b58f38da3a0646a5a5cc
Theme exporting should have been removed via 0b07f467bacfef02f3cef151befec2637565d20f

### CODE REVIEW

**Comment by darealshinji** [comment]
Didn't know about Shellcheck. I've added a few comments, please take a look at those notes. I guess this PR would also make https://github.com/KittyKatt/screenFetch/pull/8 finally obsolete.

**Comment by stevestock** [comment]
@darealshinji where can I find your notes?

**Comment by darealshinji** [comment]
They should be above [this](https://github.com/KittyKatt/screenFetch/pull/561#issuecomment-393504275) post.

**Comment by stevestock** [comment]
Are you sure you submitted the comments? [see this stackoverflow answer](https://stackoverflow.com/a/45336490)

Otherwise it’s a bug on mobile, I can check when i get home

**Comment by darealshinji** [comment]
Yeah, that was the issue.

File: screenfetch-dev [hunk 1]
```diff
@@ -458,12 +447,7 @@ done
 
 if [[ "$overrideOpts" ]]; then
 	verboseOut "Found 'o' flag in syntax. Overriding some script variables..."
-	OLD_IFS="$IFS"
-	IFS=";"
-	for overopt in "${overrideOpts}"; do
 		eval "${overrideOpts}"
-	done
-	IFS="$OLD_IFS"
```
[the remark this answers is not in the document]
Comment (stevestock) [reply screenfetch-dev]: What's the point of splitting each statement up and running them separately?  For instance, the command `screenfetch -o 'distro="debian";distro="ubuntu"'`... you *could* split it up and run each statement separately... but won't it always also evaluate as-is?  Or are there cases where the user would have a different default IFS, and it wouldn't run properly?

File: screenfetch-dev [hunk 2]
```diff
@@ -1298,17 +1284,17 @@ detectpkgs () {
 		;;
 		'Cygwin')
 			cygfix=2
-			pkgs=$(($(cygcheck -cd | wc -l) - ${cygfix}))
+			pkgs=$(($(cygcheck -cd | wc -l) - cygfix))
```
[the remark this answers is not in the document]
Comment (stevestock) [reply screenfetch-dev]: In my experience, that would make it a [magic number](https://stackoverflow.com/questions/47882/what-is-a-magic-number-and-why-is-it-bad), which I typically consider bad practice

File: screenfetch-dev [hunk 3]
```diff
@@ -1422,7 +1408,7 @@ detectcpu () {
 			cpu_mhz=$(awk -F':' '/cpu MHz/{ print int($2+.5) }' /proc/cpuinfo | head -n 1)
 		fi
 		if [ -n "$cpu_mhz" ]; then
-			if [ $(echo $cpu_mhz | cut -d. -f1) -gt 999 ]; then
+			if [ "$(echo "$cpu_mhz" | cut -d. -f1)" -gt 999 ]; then
```
[the remark this answers is not in the document]
Comment (stevestock) [reply screenfetch-dev]: I like it, making the change

File: screenfetch-dev (hunk 2 above)
[the remark this answers is not in the document]
Comment (darealshinji) [reply screenfetch-dev]: It is done for other package managers in this script too.

File: screenfetch-dev (hunk 1 above)
[the remark this answers is not in the document]
Comment (darealshinji) [reply screenfetch-dev]: No, I think you're right on this one.

File: screenfetch-dev (hunk 2 above)
[the remark this answers is not in the document]
Comment (darealshinji) [reply screenfetch-dev]: Change it. I think in this case we can live with a magic number.

[the remark this answers is not in the document]
Comment (stevestock) [reply screenfetch-dev]: done

**Comment by stevestock** [comment]
fyi, this doesn't fix all [ShellCheck](https://github.com/koalaman/shellcheck) issues, it's just a good start.

before:

    $ shellcheck -f gcc screenfetch-dev | wc -l
    687

after:

    $ shellcheck -f gcc screenfetch-dev | wc -l
    287

**Comment by darealshinji** [comment]
Yeah, I've seen that shellcheck complains a lot about this script. I don't even know if fixing all of this blindly is even the best idea.

**Comment by stevestock** [comment]
@darealshinji I tried to verify all the changes the best I could... but it's pretty tough without a test suite.  It's hard to even know which distros are needed for some code paths.  Could probably do some testing with a series of vagrant boxes... but i don't know how you'd test the more obscure distros.

**Comment by stevestock** [comment]
I'm running with shellcheck v0.5.0

**Pull request merged**